In [4]:
library(tidyverse)
players<- read_csv("players.csv")
sessions <- read_csv("sessions.csv")
head(players)
head(sessions)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimite

experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

**UBC DSCI100 individual report**

-**Author:** Yunan Deng

-**Student number:** 67274323

###**-Data description**


**-players.csv description**
number of observation is 196, and number of variables are 7.

**The name of variables are "experience", "subscribe", "hashedEmail","played_hours","name","gender"and"Age"**

- **experience:** character, indicates the players experience level.

- **subscribe:** logical, indicates whether players subcribe newsletters.(True or false)

- **hashedEmail:** character, the users hashed email number.

- **played_hours:** numerical, the number of time player play the game.

- **name:** character,the username of players.

- **gender:** character, the gender of players.

- **Age:** numerical, the age of players.


**-summary statistics(2 decimal)-mean**
 
**age**

- The average age of players is 21.14 year-old.

- The median age of players is 19 year-old.

- The range of players' age in the game is from 9 years old to 58 years old.

**played_hours**

- The average played hours of players is 5.85 hours.

- The median play time of players is 0.1 hours. (affected by "0")

- The range of play time of players from 0 to 223.1 hours

**-Potential issues**

- **N/A values:**
There are all valid cell in each columns variable, no N/A values appear, which means the average values for **Age** and **played_hours** are correct and not affected by the N/A values in Age and played_hours columns.

- **unbanlanced categoricals:** the number of male players is significant higher than other genders, which is nearly three times for females players and 5-10 times for other gender identities. This imbanlance categoricals collecting will cause bias in the prediction. Additionally, most players are Amateur level, which nearly two times number for Beginer level, indicating another potential imbanlance in the experience level distribution.

- **small sample size:** the number of rows only 196, which means only 196 players information at here. The sample size is really small, limiting the generalization for final conclusion and prediction.

- **extrem value:** the minimum value of played_hours appears "0", indicating that some players have 0 play hours, which may indicate they registered but never played. This maybe will affect the average value for played_houres lower then actually.

**-How the data were collect**

- 

In [5]:
number_rows <- nrow(players)
number_rows
number_columns <- ncol(players)
number_columns

[1] 196

[1] 7

In [6]:
statical_value_Age<-players|>summarise(
    mean_age = round(mean(Age,na.rm = TRUE), 2),
    median_age = round(median(Age,na.rm = TRUE), 2),
    min_age = round(min(Age,na.rm = TRUE), 2),
    max_age = round(max(Age,na.rm = TRUE), 2)
    
)
statical_value_Age

statical_value_played_hours<-players|>summarise(
    mean_played_hours =round(mean(played_hours,na.rm= TRUE),2),
    median_played_hours =round(median(played_hours,na.rm= TRUE),2),
    min_played_hours =round(min(played_hours,na.rm= TRUE),2),
    max_played_hours =round(max(played_hours,na.rm= TRUE),2)
)

statical_value_played_hours


mean_age median_age min_age max_age
1 21.14    19         9       58

mean_played_hours median_played_hours min_played_hours max_played_hours
1 5.85              0.1                 0                223.1

In [7]:
is.na(players)

experience subscribe hashedEmail played_hours name  gender Age  
 [1,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
 [2,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
 [3,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
 [4,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
 [5,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
 [6,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
 [7,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
 [8,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
 [9,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[10,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[11,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[12,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[13,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[14,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[15,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[16,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[17,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[18,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[19,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[20,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[21,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[22,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[23,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[24,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[25,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[26,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[27,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[28,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[29,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[30,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[31,] ⋮          ⋮         ⋮           ⋮            ⋮     ⋮      ⋮    
[32,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[33,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[34,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[35,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[36,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[37,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[38,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[39,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[40,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[41,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[42,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[43,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[44,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[45,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[46,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[47,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[48,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[49,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[50,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[51,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[52,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[53,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[54,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[55,] FALSE      FALSE     FALSE       FALSE        FALSE FALSE  FALSE
[56,] FALSE      FALSE     FAL

In [8]:
library(dplyr)
number_for_experience_level <- players|>
count(experience)
number_for_experience_level

number_for_gender <- players|>
count(gender)
number_for_gender

experience n 
1 Amateur    63
2 Beginner   35
3 Pro        14
4 Regular    36
5 Veteran    48

gender            n  
1 Agender             2
2 Female             37
3 Male              124
4 Non-binary         15
5 Other               1
6 Prefer not to say  11
7 Two-Spirited        6

**sessions.csv description**
- **total number of rows is 1535**

- **total number of variables is 5**

-**the name of varables are original_end_time, original_start_time, end_time, start_time, and hashedEmaile.**

- **original_end_time:** numeric, the original end time of player without translation to date/month/year/time.

- **original_start_time:** numeric, the original start time of player without translation to date/month/year/time.

- **end_time:** character, the end time of player.

- **start_time:** character,the start time of player.

- **hashedEmaile:** character, the users hashed email number.

-**summary statistics(2 decimal)**

- **original_end_time**

- the average original end time for each players we collect is 1.719201e+12.


- the median orginal end time is1.7192e+12.

- the range of players' original end time is from1.7124e+12	to 1.72733e+12.

- **original_start_time**

- the average original end time for each players we collect is 1.719196e+12.

- the median orginal end time is 1.71918e+12.

- the range of players' original end time is from 1.7124e+12 to 1.72734e+12.


- **pottential issue**

- no missing data

- There are a lot of **zero value**, which means the start time for players and ends time for players are same, the game duration is 0. The number is great(1104), which may affect the prediction. I think maybe is the static lagging problem.

In [10]:
number_rows_s <- nrow(sessions)
number_cols_s <- ncol(sessions)
number_rows_s
number_cols_s

[1] 1535

[1] 5

In [11]:
statical_value_original_start<-sessions|>summarise(
    mean_start = round(mean(original_start_time,na.rm = TRUE), 2),
    median_start = round(median(original_start_time,na.rm = TRUE), 2),
    min_start = round(min(original_start_time,na.rm = TRUE), 2),
    max_start = round(max(original_start_time,na.rm = TRUE), 2)
    
)
statical_value_original_start

statical_value_original_end<-sessions|>summarise(
    mean_end =round(mean(original_end_time,na.rm= TRUE),2),
    median_end =round(median(original_end_time,na.rm= TRUE),2),
    min_end =round(min(original_end_time,na.rm= TRUE),2),
    max_end =round(max(original_end_time,na.rm= TRUE),2)
)

statical_value_original_end

mean_start   median_start min_start  max_start  
1 1.719201e+12 1.7192e+12   1.7124e+12 1.72733e+12

mean_end     median_end  min_end    max_end    
1 1.719196e+12 1.71918e+12 1.7124e+12 1.72734e+12

In [12]:
is.na(sessions)

hashedEmail start_time end_time original_start_time original_end_time
 [1,] FALSE       FALSE      FALSE    FALSE               FALSE            
 [2,] FALSE       FALSE      FALSE    FALSE               FALSE            
 [3,] FALSE       FALSE      FALSE    FALSE               FALSE            
 [4,] FALSE       FALSE      FALSE    FALSE               FALSE            
 [5,] FALSE       FALSE      FALSE    FALSE               FALSE            
 [6,] FALSE       FALSE      FALSE    FALSE               FALSE            
 [7,] FALSE       FALSE      FALSE    FALSE               FALSE            
 [8,] FALSE       FALSE      FALSE    FALSE               FALSE            
 [9,] FALSE       FALSE      FALSE    FALSE               FALSE            
[10,] FALSE       FALSE      FALSE    FALSE               FALSE            
[11,] FALSE       FALSE      FALSE    FALSE               FALSE            
[12,] FALSE       FALSE      FALSE    FALSE               FALSE            
[13,] FALSE       FALSE      FALSE    FALSE               FALSE            
[14,] FALSE       FALSE      FALSE    FALSE               FALSE            
[15,] FALSE       FALSE      FALSE    FALSE               FALSE            
[16,] FALSE       FALSE      FALSE    FALSE               FALSE            
[17,] FALSE       FALSE      FALSE    FALSE               FALSE            
[18,] FALSE       FALSE      FALSE    FALSE               FALSE            
[19,] FALSE       FALSE      FALSE    FALSE               FALSE            
[20,] FALSE       FALSE      FALSE    FALSE               FALSE            
[21,] FALSE       FALSE      FALSE    FALSE               FALSE            
[22,] FALSE       FALSE      FALSE    FALSE               FALSE            
[23,] FALSE       FALSE      FALSE    FALSE               FALSE            
[24,] FALSE       FALSE      FALSE    FALSE               FALSE            
[25,] FALSE       FALSE      FALSE    FALSE               FALSE            
[26,] FALSE       FALSE      FALSE    FALSE               FALSE            
[27,] FALSE       FALSE      FALSE    FALSE               FALSE            
[28,] FALSE       FALSE      FALSE    FALSE               FALSE            
[29,] FALSE       FALSE      FALSE    FALSE               FALSE            
[30,] FALSE       FALSE      FALSE    FALSE               FALSE            
[31,] ⋮           ⋮          ⋮        ⋮                   ⋮                
[32,] FALSE       FALSE      FALSE    FALSE               FALSE            
[33,] FALSE       FALSE      FALSE    FALSE               FALSE            
[34,] FALSE       FALSE      FALSE    FALSE               FALSE            
[35,] FALSE       FALSE      FALSE    FALSE               FALSE            
[36,] FALSE       FALSE      FALSE    FALSE               FALSE            
[37,] FALSE       FALSE      FALSE    FALSE               FALSE            
[38,] FALSE       FALSE      FALSE    FALSE               FALSE            
[39,] FALSE       FALSE      FALSE    FALSE               FALSE            
[40,] FALSE       FALSE      FALSE    FALSE               FALSE            
[41,] FALSE       FALSE      FALSE    FALSE               FALSE            
[42,] FALSE       FALSE      FALSE    FALSE               FALSE            
[43,] FALSE       FALSE      FALSE    FALSE               FALSE            
[44,] FALSE       FALSE      FALSE    FALSE               FALSE            
[45,] FALSE       FALSE      FALSE    FALSE               FALSE            
[46,] FALSE       FALSE      FALSE    FALSE               FALSE            
[47,] FALSE       FALSE      FALSE    FALSE               FALSE            
[48,] FALSE       FALSE      FALSE    FALSE               FALSE            
[49,] FALSE       FALSE      FALSE    FALSE               FALSE            
[50,] FALSE       FALSE      FALSE    FALSE               FALSE            
[51,] FALSE       FALSE      FALSE    FALSE               FALSE            
[52,] FALSE       FALSE      FALSE    FALSE           

In [21]:
zero_issue <- sessions|>
filter(original_start_time == original_end_time)

count(zero_issue)




n   
1 1104

**where we collect the data?**

- Our data are sourced from Minecraft werver which establied by one research group from UBC Department of Computer Science, leaded by the Professor Fanch Mood. They capture the actions of players and the informations.
- Some informations, like hasedEmail, are encrypted.
- Original_start_time and original_end_time can transfer a form we can directly reading, start_time and end_time.
